### PR TITLE
Updating to fix the issue where bash does not inherit the cwd correctly

### PR DESCRIPTION
**Pull request by ksubrama**
\cc @tyler-ball @schisamo @ksubrama

### CODE REVIEW

File: resources/msys2_package.rb [hunk 1]
```diff
@@ -76,30 +84,26 @@ def msys2_init
...
-      msys2_exec('sync msys2 packages', 'pacman -Sy')
-      msys2_exec('upgrade msys2 core packages', 'pacman -S --noconfirm --needed bash pacman msys2-runtime msys2-runtime-devel')
-      msys2_exec('upgrade entire msys2 system', 'pacman -Suu --noconfirm')
-      # Sometimes this step seems to be necessary.  Unsure why..
-      # Not doing this results in only a few packages getting upgraded.
-      msys2_exec('upgrade entire msys2 system', 'pacman -Syu --noconfirm')
-      file win_friendly_path(updated_marker_file) do
-        action :touch
-      end
+    if ::File.exist?(::File.join(root, 'usr/bin/update-core'))
```
Comment (tyler-ball) [inline resources/msys2_package.rb]: `usr/bin/update-core` doesn't get removed so this block executes every time and re-installs a bunch of stuff

Comment (tyler-ball) [inline resources/msys2_package.rb]: Also saw this output the second time I ran:

```
                  warning: bash-4.3.042-4 is up to date -- skipping
                  warning: pacman-5.0.0.6348.cc5a8f1-1 is up to date -- skipping
                  warning: msys2-runtime-2.4.1.16860.40c26fc-1 is up to date -- skipping
                  warning: msys2-runtime-devel-2.4.1.16860.40c26fc-1 is up to date -- skipping
```

But I know there is a pacman 5.0.1...

**Comment by schisamo** [comment]
We need to find a way to make all of this more idepodent. I see lots of output like the following when multiple CCRs are run:

```
:: Processing package changes...
reinstalling gettext...
reinstalling asciidoc...
reinstalling gawk...
reinstalling m4...
reinstalling diffutils...
reinstalling autoconf...
reinstalling autoconf2.13...
reinstalling autogen...
reinstalling perl...
reinstalling automake1.6...
reinstalling automake1.7...
reinstalling automake1.8...
reinstalling automake1.9...
reinstalling automake1.10...
reinstalling automake1.11...
reinstalling automake1.12...
reinstalling automake1.13...
reinstalling automake1.14...
reinstalling automake1.15...
reinstalling automake-wrapper...
reinstalling bison...
reinstalling diffstat...
reinstalling dos2unix...
reinstalling file...
reinstalling flex...
reinstalling gdb...
reinstalling gperf...
reinstalling grep...
reinstalling groff...
reinstalling help2man...
reinstalling intltool...
reinstalling lemon...
reinstalling libtool...
reinstalling libunrar...
reinstalling make...
reinstalling man-db...
reinstalling sed...
reinstalling pacman...
reinstalling patch...
reinstalling patchutils...
reinstalling pkg-config...
reinstalling pkgfile...
reinstalling rcs...
reinstalling scons...
reinstalling swig...
reinstalling texinfo...
reinstalling texinfo-tex...
reinstalling unrar...
reinstalling wget...
reinstalling xmlto...

<SNIP>

:: Processing package changes...
reinstalling mingw-w64-i686-binutils...
reinstalling mingw-w64-i686-headers-git...
reinstalling mingw-w64-i686-crt-git...
reinstalling mingw-w64-i686-libwinpthread-git...
reinstalling mingw-w64-i686-gcc-libs...
reinstalling mingw-w64-i686-winpthreads-git...
reinstalling mingw-w64-i686-gcc...
reinstalling mingw-w64-i686-gcc-ada...
reinstalling mingw-w64-i686-gcc-libgfortran...
reinstalling mingw-w64-i686-gcc-fortran...
reinstalling mingw-w64-i686-gcc-objc...
reinstalling mingw-w64-i686-gdb...
reinstalling mingw-w64-i686-libmangle-git...
reinstalling mingw-w64-i686-make...
reinstalling mingw-w64-i686-pkg-config...
reinstalling mingw-w64-i686-tools-git...
```

You should be able to do some sort of check with `pacman` to check if things are already installed.

**Comment by tyler-ball** [comment]
We're going to go ahead and release this, then come back later to fix the long Chef runs

File: resources/msys2_package.rb (hunk 1 above)
Comment (ksubrama) [inline resources/msys2_package.rb]: So the initial step is to update pacman to a version that then supports just doing Syuu and then rerunning that step but there's some magic to it that I haven't figured out.  When I do it on the console, it deletes update-core and upgrades a number of other packages...  but *shrug*.  It's ok for now.

**Pull request merged**